keybase/platform_darwin.go

```diff
@@ -183,6 +183,12 @@ func (c context) stop() error {
 		c.log.Warningf("Error requesting app exit: %s", appExitErr)
 	}
 
+	// Stop the redirector so it can be upgraded for all users.
```

Comment:
It looks like you're uninstalling the redirector first, not just stopping it. That's fine, but the comments and error messages should reflect what's actually happening.

Also, what happens if the upgrade fails?

Reply:
One more question: will this work before a redirector exists?

Reply:
> It looks like you're uninstalling the redirector first, not just stopping it. That's fine, but the comments and error messages should reflect what's actually happening.

Under the covers, `--uninstall-redirector` just calls `stopRedirector` in the root helper.  It's following the naming convention of everything else, including `--uninstall-kbfs` which just stops KBFS but doesn't actually remove anything from the system.

> Also, what happens if the upgrade fails?

Then the redirector will remain shut down until the next time the app starts up.  I think this matches the behavior of shutting down the rest of the application before the upgrade happens.

> One more question: will this work before a redirector exists?

Nope. That's why I mention in the PR description that I won't merge this until the redirector changes are merged into master.